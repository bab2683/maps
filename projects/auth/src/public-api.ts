/*
 * Public API Surface of auth
 */

export * from './lib/auth.model';
export * from './lib/auth.module';
export * from './lib/store';
