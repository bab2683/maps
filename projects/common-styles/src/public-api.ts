/*
 * Public API Surface of common-styles
 */
export * from './lib/common-styles.module';
