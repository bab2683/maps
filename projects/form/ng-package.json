{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/@bab/form",
  "lib": {
    "entryFile": "src/public-api.ts"
  }
}
