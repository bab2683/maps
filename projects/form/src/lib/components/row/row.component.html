<div class="row" [ngClass]="classes">
  <ng-content></ng-content>
</div>
