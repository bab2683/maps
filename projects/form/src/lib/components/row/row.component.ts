import { Component, Input, OnInit } from '@angular/core';

import { FormRow } from '../../models';

@Component({
  selector: 'bab-form-row',
  templateUrl: './row.component.html',
  styleUrls: ['./row.component.scss']
})
export class RowComponent implements OnInit {
  @Input() public data: FormRow;

  public classes: string[];

  public ngOnInit(): void {
    this.classes = this.createClasses(this.data);
  }

  private createClasses(data: FormRow): string[] {
    const { fields, sizes } = data;
    if (sizes) {
      return Object.entries(sizes).reduce((result, current) => {
        result.push(`row--${current[0]}${current[1]}`);
        return result;
      }, []);
    }
    return [`row--b${fields.length}`];
  }
}
