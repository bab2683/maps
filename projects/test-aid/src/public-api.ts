/*
 * Public API Surface of test-aid
 */

export * from './lib/angular-fire.mock';
export * from './lib/rxjs.mock';
export * from './lib/utils.mock';
