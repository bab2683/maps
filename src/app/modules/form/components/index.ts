export * from './composer';
export * from './field';
export * from './input';
