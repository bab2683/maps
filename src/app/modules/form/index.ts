export * from './form.module';
export * from './interfaces';
