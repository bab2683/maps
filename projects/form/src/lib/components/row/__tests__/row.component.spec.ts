import { NO_ERRORS_SCHEMA } from '@angular/compiler/src/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { FormRow } from '@mod/form';

import { RowComponent } from '../row.component';

describe('RowComponent', () => {
  let component: RowComponent;
  let fixture: ComponentFixture<RowComponent>;

  const testData: FormRow = {
    fields: [
      {
        label: 'Your email',
        name: 'email',
        type: 'email'
      },
      {
        label: 'Your password',
        name: 'password',
        type: 'password'
      }
    ]
  };

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [RowComponent],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(RowComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('onInit', () => {
    it('should create the classes according to content only', () => {
      component.data = testData;
      fixture.detectChanges();
      component.ngOnInit();

      expect(component.classes).toBeDefined();
      expect(component.classes).toEqual(['row--b2']);
    });
    it('should create the classes according to content only', () => {
      const modifiedData: FormRow = {
        ...testData,
        sizes: {
          s: 1
        }
      };

      component.data = modifiedData;
      fixture.detectChanges();
      component.ngOnInit();

      expect(component.classes).toBeDefined();
      expect(component.classes).toEqual(['row--s1']);
    });
  });
});
