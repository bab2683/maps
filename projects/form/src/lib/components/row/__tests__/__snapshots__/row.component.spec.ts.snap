// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowComponent should match snapshot 1`] = `
<form-row>
  <div
    class="row"
  />
</form-row>
`;
